src/urllib3/request.py

```diff
@@ -161,7 +174,7 @@ def request_encode_body(
                 )
             else:
                 body, content_type = (
-                    urlencode(fields),
+                    urlencode(fields),  # type: ignore
```

Comment:
`error: Argument 1 to "urlencode" has incompatible type "Union[Sequence[Union[Tuple[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]], RequestField]], Dict[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]]]"; expected "Union[Mapping[Any, Any], Mapping[Any, Sequence[Any]], Sequence[Tuple[Any, Any]], Sequence[Tuple[Any, Sequence[Any]]]]"
`

Reply:
> Since encoding into the URL only actually supports Union[Sequence[Tuple[str, Union[str, bytes]]], Mapping[str, Union[str, bytes]]] let's use this as our type hint for fields of request_encode_url?


If we do this, then we will face with error in [encode_multipart_formdata](https://github.com/urllib3/urllib3/blob/ed23062eef4900f4d4c7540f51062c94bb9dcf07/src/urllib3/request.py#L159)

`
error: Argument 1 to "encode_multipart_formdata" has incompatible type "Union[Sequence[Tuple[str, Union[str, bytes]]], Mapping[str, Union[str, bytes]]]"; expected "Union[Sequence[Union[Tuple[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]], RequestField]], Dict[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]]]"
`

because `encode_multipart_formdata` uses the [_TYPE_FIELDS in filepost.py](https://github.com/urllib3/urllib3/blob/ed23062eef4900f4d4c7540f51062c94bb9dcf07/src/urllib3/filepost.py#L14).
Should we ignore this?

Reply:
`_TYPE_FIELDS` should be allowed in `request_encode_body`, add a `type: ignore` to the `urlencode()` if `encode_multipart` is false